File: java-version/src/test/java/com/github/leetcode/LC2502/AllocatorTest.java
package com.github.leetcode.LC2502;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AllocatorTest {
    @Test
    public void TestAllocator() {
        Allocator allocator = new Allocator(10);
        System.out.println(allocator.allocate(1, 1));
        System.out.println(allocator.allocate(1, 2));
        System.out.println(allocator.allocate(1, 3));
        System.out.println(allocator.freeMemory(2));
        System.out.println(allocator.allocate(3, 4));
        System.out.println(allocator.allocate(1, 1));
        System.out.println(allocator.allocate(1, 1));
        System.out.println(allocator.freeMemory(1));
        System.out.println(allocator.allocate(10, 2));
        System.out.println(allocator.freeMemory(7));
    }
}

File: python-version/LC2502/allocator_test.py
import unittest
from allocator import Allocator

class AllocatorTest(unittest.TestCase):
    def test_allocator(self):
        allocator = Allocator(10)
        print(allocator.allocate(1, 1))
        print(allocator.allocate(1, 2))
        print(allocator.allocate(1, 3))
        print(allocator.freeMemory(2))
        print(allocator.allocate(3, 4))
        print(allocator.allocate(1, 1))
        print(allocator.allocate(1, 1))
        print(allocator.freeMemory(1))
        print(allocator.allocate(10, 2))
        print(allocator.freeMemory(7))

if __name__ == '__main__':
    unittest.main()